pkg/scheduler/internal/cache/cache_test.go

```diff
@@ -1539,6 +1539,180 @@ func compareCacheWithNodeInfoSnapshot(cache *schedulerCache, snapshot *Snapshot)
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "node-6",
+				Labels: map[string]string{
+					v1.LabelZoneRegion:        "region-2",
+					v1.LabelZoneFailureDomain: "zone-2",
+				},
+			},
+		},
+		// Node 7: a node with proper region and zone labels in a region and zone
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "node-7",
+				Labels: map[string]string{
+					v1.LabelZoneRegion:        "region-2",
+					v1.LabelZoneFailureDomain: "zone-2",
+				},
+			},
+		},
+	}
+
+	var cache *schedulerCache
+	var snapshot *Snapshot
+	type operation = func()
+
+	addNode := func(i int) operation {
+		return func() {
+			if err := cache.AddNode(nodes[i]); err != nil {
+				t.Error(err)
+			}
+			_, ok := snapshot.nodeInfoMap[nodes[i].Name]
+			if !ok {
+				snapshot.nodeInfoMap[nodes[i].Name] = cache.nodes[nodes[i].Name].info
+			}
+		}
+	}
+
+	updateSnapshot := func() operation {
+		return func() {
+			cache.updateNodeInfoSnapshotList(snapshot, true)
+			if err := compareCacheWithNodeInfoSnapshot(cache, snapshot); err != nil {
+				t.Error(err)
+			}
+		}
+	}
+
+	tests := []struct {
+		name       string
+		operations []operation
+		expected   []string
+	}{
+		{
+			name:       "Empty cache",
+			operations: []operation{},
+			expected:   []string{},
+		},
+		{
+			name:       "Single node",
+			operations: []operation{addNode(0)},
+			expected:   []string{"node-0"},
+		},
+		{
+			name:       "Two nodes",
+			operations: []operation{addNode(0), updateSnapshot(), addNode(1)},
+			expected:   []string{"node-0", "node-1"},
+		},
+		{
+			name:       "bug 91601, two nodes, update the snapshot and add two nodes in different zones",
+			operations: []operation{addNode(2), addNode(3), updateSnapshot(), addNode(4), addNode(0)},
+			expected:   []string{"node-2", "node-0", "node-3", "node-4"},
+		},
+		{
+			name:       "bug 91601, 6 nodes, one in a different zone",
+			operations: []operation{addNode(2), addNode(3), addNode(4), addNode(5), updateSnapshot(), addNode(6), addNode(0)},
+			expected:   []string{"node-2", "node-0", "node-3", "node-4", "node-5", "node-6"},
+		},
+		{
+			name:       "bug 91601, 7 nodes, two in a different zone",
+			operations: []operation{addNode(2), updateSnapshot(), addNode(3), addNode(4), updateSnapshot(), addNode(5), addNode(6), addNode(0), addNode(1)},
+			expected:   []string{"node-2", "node-0", "node-3", "node-1", "node-4", "node-5", "node-6"},
+		},
+		{
+			name:       "bug 91601, 7 nodes, two in a different zone, different zone order",
+			operations: []operation{addNode(2), addNode(1), updateSnapshot(), addNode(3), addNode(4), updateSnapshot(), addNode(5), addNode(6), addNode(0)},
+			expected:   []string{"node-2", "node-1", "node-3", "node-0", "node-4", "node-5", "node-6"},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			cache = newSchedulerCache(time.Second, time.Second, nil)
+			snapshot = NewEmptySnapshot()
+
+			for _, op := range test.operations {
+				op()
+			}
```

Comment:
With the previous 2 nits, the proposed changes is like this: [diff](https://gist.github.com/Huang-Wei/102ce8ba61d7ec61bfc61a9aac7c07cd)